### DISCUSSION TITLE
S3 adapter configuration

### DISCUSSION THREAD

**Discussion by AlessandroMinoccheri** [directus/directus#5676] [Archive]
Hi all, I can't find the documentation about s3 storage for directus9.
Can someone explains to me how to configure into `.env` file an s3 bucket to store my files?

Thanks!

**Comment by rijkvanzanten** [comment]
https://docs.directus.io/guides/api-config/#file-storage and https://docs.directus.io/reference/environment-variables/#s3-s3 are the docs around this 🙂

**Reply by AlessandroMinoccheri** [reply]
I tried this but without success: 

```
STORAGE_LOCATIONS="aws"

STORAGE_AWS_DRIVER: "s3"
STORAGE_AWS_KEY: "KEY"
STORAGE_AWS_SECRET: "XXX"
STORAGE_AWS_BUCKET: "bucket"
STORAGE_AWS_REGION: "eu-west-1"
```


I get this error:

```
{
  "errors": [
    {
      "message": "E_INVALID_CONFIG: Make sure to define driver for aws disk",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

**Reply by rijkvanzanten** [reply]
Where are you setting those env vars? Locations has a `=`, but the others `:`. Is this in a docker environment?

**Reply by AlessandroMinoccheri** [reply]
In .env was with `=` the problem was the cache!

Now it works fine!

Thank @rijkvanzanten

**Reply by Sleepful** [reply]
those links to the docs are broken now, use this one:
- https://docs.directus.io/self-hosted/config-options.html#file-storage
